server/common/EntryParser.java

```diff
@@ -0,0 +1,279 @@
+                    (yaml) -> yaml.isPrimitive() && yaml.asPrimitive().isFloat(),
+                    (yaml) -> yaml.asPrimitive().asFloat(),
+                    "<float>"
+            );
+            static final Leaf<Boolean> BOOLEAN = new Leaf<>(
+                    (yaml) -> yaml.isPrimitive() && yaml.asPrimitive().isBoolean(),
+                    (yaml) -> yaml.asPrimitive().asBoolean(),
+                    "<boolean>"
+            );
+            static final Leaf<List<String>> LIST_STRING = new Leaf<>(
+                    (yaml) -> yaml.isList() && iterate(yaml.asList()).allMatch(elem -> elem.isPrimitive() && elem.asPrimitive().isString()),
+                    (yaml) -> iterate(yaml.asList()).map(elem -> elem.asPrimitive().asString()).toList(),
+                    "<[string, ...]>");
+            static final Leaf<List<Integer>> LIST_INTEGER = new Leaf<>(
+                    (yaml) -> yaml.isList() && iterate(yaml.asList()).allMatch(elem -> elem.isPrimitive() && elem.asPrimitive().isInt()),
+                    (yaml) -> iterate(yaml.asList()).map(elem -> elem.asPrimitive().asInt()).toList(),
+                    "<[integer, ...]>");
+            static final Leaf<List<Float>> LIST_FLOAT = new Leaf<>(
+                    (yaml) -> yaml.isList() && iterate(yaml.asList()).allMatch(elem -> elem.isPrimitive() && elem.asPrimitive().isFloat()),
+                    (yaml) -> iterate(yaml.asList()).map(elem -> elem.asPrimitive().asFloat()).toList(),
+                    "<[float, ...]>");
+            static final Leaf<List<Boolean>> LIST_BOOLEAN = new Leaf<>(
+                    (yaml) -> yaml.isList() && iterate(yaml.asList()).allMatch(elem -> elem.isPrimitive() && elem.asPrimitive().isBoolean()),
+                    (yaml) -> iterate(yaml.asList()).map(elem -> elem.asPrimitive().asBoolean()).toList(),
+                    "<[boolean, ...]>");
+
+            private final Function<Yaml, Boolean> validator;
+            private final Function<Yaml, T> converter;
+            private final String help;
+
+            private Leaf(Function<Yaml, Boolean> validator, Function<Yaml, T> converter, String help) {
+                this.validator = validator;
+                this.converter = converter;
+                this.help = help;
+            }
+
+            @Override
+            T parse(Yaml yaml, String scope) {
+                if (!validator.apply(yaml)) {
+                    throw TypeDBException.of(CONFIG_OPTION_UNEXPECTED_VALUE_TYPE, scope, yaml, help);
+                } else {
+                    return converter.apply(yaml);
+                }
+            }
+
+            @Override
+            boolean isLeaf() {
+                return true;
+            }
+
+            @Override
+            Leaf<T> asLeaf() {
+                return this;
+            }
+
+            public String help() {
+                return help;
+            }
+        }
+    }
+
+    static class Unnamed<TYPE> extends EntryParser {
```

Comment:
This is the thing PicoCLI didn't support: we want to be able to have the user use a dynamic name for a _key_ in our yaml config - that's what this type of entry does for us.